File: fullstack/app/Service/MethodService/Transfers/UTARequest.php
<?php

namespace App\Service\MethodService\Transfers;

use App\Service\MethodService\Transfers\Electre1s\VariantDTO;

class UTARequest
{
    public array $alternativesIndifferences = [[]];

    public array $alternativesPreferences = [[]];

    /** @var int[] */
    public ?array $alternativesRanks = [];

    /** @var int[] */
    public array $criteriaNumberOfBreakPoints = [];


    /** @var string[] */
    public array $colnamesPerformanceTable = [];

    /** @var string[] */
    public array $criteriaMinMax = [];

    public float $epsilon = 0.0;

    /** @var int[] */
    public array $performanceTable = [];

    /** @var string[] */
    public array $rownamesPerformanceTable = [];


}

File: backend/src/main/kotlin/pl/poznan/put/decision_support/decisionmethods/model/domain/uta/request/UTARequest.kt
package pl.poznan.put.decision_support.decisionmethods.model.domain.uta.request

data class UTARequest(
    val alternativesIndifferences: List<List<String>>? = null,
    val alternativesPreferences: List<List<String>>? = null,
    val alternativesRanks: List<Int>? = null,
    val colnamesPerformanceTable: List<String?>,
    val criteriaMinMax: List<String?>,
    val criteriaNumberOfBreakPoints: List<Int?>,
    val epsilon: Double,
    val performanceTable: List<List<Float?>>,
    val rownamesPerformanceTable: List<String?>
)